### ISSUE TITLE
A11y: External follow button generates an iframe with no "title" attribute

### ISSUE THREAD

**Issue by jp-imagines** [Automattic/wp-calypso#74220]
### Quick summary

For accessibility reasons, iframes/frames should have a `title` attribute to assist people using screen readers: https://dequeuniversity.com/tips/provide-iframe-titles

The "external follow button" uses HTML/JS code which generates an iframe on the site without a `title` attribute. Since the iframe isn't inserted directly, there's also no way to add a `title` attribute manually.

### Steps to reproduce

1. Follow the guide to create an external follow button: https://wordpress.com/support/follow-button/
  * Generator is here: https://developer.wordpress.com/docs/follow-button-creation/
2. Add the external follow button to an AT site.
3. View the live site and/or run a Page Speed analysis: https://pagespeed.web.dev/ and check the accessibility section.

### What you expected to happen

The generated iframe should have a `title` attribute.

### What actually happened

There's no `title`. Accessibility reports may warn about this a11y concern.

![Markup on 2023-03-08 at 11:57:17](https://user-images.githubusercontent.com/5630717/223778559-e52a883c-bdfd-4f02-b06b-590fa977c48c.png)

### Impact

Some (< 50%)

### Available workarounds?

Yes, easy to implement

### Platform (Simple and/or Atomic)

Atomic

### Logs or notes

Reported in 6020986-zd-woothemes.

Switching to the Subscribe block is a potential workaround for those who aren't tied to the smaller footprint of the external follow button (and who use the block editor).

**Comment by kavyagokul** [comment]
📌 SCRUBBING : RESULT
- Tested on AT - Replicated 
- Tested on Self-hosted - Replicated 
- Replicable on Core - Yes

📌 ACTIONS
- Marked as Triaged for Quality Squad review

**Comment by davemart-in** [comment]
PR is up D110939-code

**Comment by davemart-in** [comment]
Should be fixed.
